component-lifecycle/src/components/Inputs.tsx

```diff
@@ -0,0 +1,121 @@
+import { TextField, Grid, MenuItem } from "@mui/material";
+import React, { useEffect, useState } from "react";
+
+const selectOption = [
+  {
+    value: "Male",
+    label: "Male",
+  },
+  {
+    value: "Female",
+    label: "Female",
+  },
+];
+
+const Input = () => {
+  const [name, setName] = useState<string>("");
+  const [age, setAge] = useState<number>(0);
+  const [gender, setGender] = useState<string>("");
+  const [image, setImage] = useState<string>("/assets/child.jpg");
+  const [profile, setProfile] = useState<string>("");
+
+  useEffect(() => {
+    age < 20
+      ? setImage("/assets/child.jpg")
+      : age < 60
+      ? setImage("/assets/adult.jpg")
+      : setImage("/assets/elder.jpeg");
+    gender.includes("Male")
+      ? setProfile("/assets/adult.jpg")
+      : gender.includes("Female")
+      ? setProfile("/assets/women.jpeg")
+      : setProfile("");
```

Comment:
can we refactor this into a function and call that function inside the use effect